### PR TITLE
feat(integrations): Django Slack integrations

### PR DESCRIPTION
**Pull request by Sharonsyra**
### What does this PR do?
• Notification is sent to slack on the allocation of an asset.

### Description of tasks to be completed
• A notification is sent to user's slack when assigned an asset
• A notification is sent to an admin when available assets exceed the limit

### What are the relevant pivotal tracker stories
• [#157183403](https://www.pivotaltracker.com/story/show/157183403)

### Screenshots

Warning message sent to the OPS channel when the number of assets of model number 23 is below 10

screen shot 2018-06-20 at 16 10 47

Allocation message to asset's new current owner

screen shot 2018-06-20 at 16 11 00

### CODE REVIEW

**Comment by collinmutembei** [comment]
@Sharonsyra Fix the failing tests

**Comment by collinmutembei** [comment]
@precious-ijege What's the hold on fixing the tests?

**Comment by precious-ijege** [comment]
@collinmutembei still working on it, so many tests are failing.

File: core/slack_bot.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    """Slack Integration class"""
+
+    def __init__(self):
+        """set the slack token"""
+        self.slack_client = SlackClient(os.getenv('SLACK_TOKEN'))
+
+    def get_user_slack_id(self, user_email):
+        """Get the slack user ID using the user email"""
+        response = self.slack_client.api_call("users.list")
+        users = response["members"]
```
Comment (collinmutembei) [inline core/slack_bot.py]: The tests are failing as a result of a `KeyError: 'members'` every time an `AssetStatus` object is saved in the database (there's a `post_save` receiver on that model) https://github.com/AndelaOSP/art-backend/blob/0d8e9a71b4cdae1c296ed0840f4e83f17f53addb/core/models/asset.py#L299-L310

One more thing to keep in mind is that the changes on this PR require an environment variable called `SLACK_TOKEN` to initialise the `SlackIntegration` class. Could be that this variable is not set on circleci.

Comment (collinmutembei) [reply core/slack_bot.py]: Also, update the [README](https://github.com/AndelaOSP/art-backend#set-up-environment-variables) to reflect the environment variables that need to be set.

File: api/tests/test_asset_api.py [hunk 2]
```diff
@@ -14,12 +14,14 @@
                          AssetSubCategory,
                          AssetCategory)
 
+from api.tests import APIBaseTestCase
 User = get_user_model()
 client = APIClient()
 
 
-class AssetTestCase(TestCase):
```
Comment (collinmutembei) [inline api/tests/test_asset_api.py]: Flake8 is complaining.
> F401 'django.test.TestCase' imported but unused

**Pull request merged**